docker-host-bridge

```diff
@@ -0,0 +1,685 @@
+
+function wg_iface_detect {
+    if sudo test -e $wg_iface_file; then
+        wg_iface=$(sudo cat $wg_iface_file)
+    fi
+}
+
+function wg_host_start {
+    wg_iface_detect
+
+    _wg_shared_generate_keys
+    _wg_host_prepare
+    _wg_host_generate_config
+    _wg_host_tunnel_create
+    _wg_host_tunnel_configure
+}
+
+function wg_host_stop {
+    wg_iface_detect
+    echo $wg_iface
+    if [[ -n "$wg_iface" ]]; then
+        sudo rm /var/run/wireguard/$wg_iface.sock
+        sudo rm $wg_iface_file
+        wg_iface=
+    fi
+}
+
+function _wg_host_prepare {
+    if [[ ! $OSTYPE =~ "darwin"* ]]; then
+        err "\`docker-host-bridge\` does not know how to prepare host of type '$OSTYPE' for use with WireGuard."
+        exit 1
+    fi
+    if ! which wireguard-go &>/dev/null; then
+        log Wireguard binary is missing.
+        if which brew &>/dev/null; then
+            run brew install wireguard-tools
+            if ! which wireguard-go &>/dev/null; then
+                err "Failed to install `wireguard-tools`. Aborting!"
+                return 1
+            fi
+            return
+        fi
+        err "Wireguard is not installed, and neither is Homebrew."
+        return 1
+    fi
+}
+
+function _wg_host_generate_config {
+    # Wireguard configuration for the macOS host.
+    log Creating a WireGuard host config
+    if [[ -e "$CONFIG_DIR"/wg-host.conf ]]; then
+        success WireGuard host config already exists.
+    else
+        cat > "$CONFIG_DIR"/wg-host.conf <<EOF
+[Interface]
+PrivateKey = $(cat "$CONFIG_DIR"/host.key)
+ListenPort = $wg_listen_port
+
+[Peer]
+PublicKey = $(cat "$CONFIG_DIR"/moby.pub)
+AllowedIPs = $wg_network, $dkr_network, 0.0.0.0/0
+PersistentKeepalive = 25
+EOF
+    fi
+}
+
+function _wg_host_tunnel_create {
+    if [[ -z "$wg_iface" ]] || ! ifconfig $wg_iface >/dev/null 2>&1; then
+        # no interface
+        log Creating WireGuard interface
+        sudo mkdir -p /var/run/wireguard
+        run sudo env WG_TUN_NAME_FILE=$wg_iface_file wireguard-go utun
+        run sudo chmod a+r $wg_iface_file
+        wg_iface=$(sudo cat $wg_iface_file)
+    else
+        success WireGuard interface $(value $wg_iface) already exists.
+    fi
+}
+
+function _wg_host_tunnel_configure {
+    log Starting WireGuard interface
+    if [[ $(ifconfig $wg_iface | grep -c -F -e UP -e "inet $wg_host") == "2" ]]; then
+      success WireGuard host interface $(value $wg_iface) is already configured.
+    else
+      log Configuring $(value $wg_iface) interface
+      run sudo wg setconf $wg_iface "$CONFIG_DIR"/wg-host.conf
+      run sudo ifconfig $wg_iface $wg_host $wg_moby up
+    fi
+}
+
+# -------- Wireguard configuration for Docker Moby Linux VM
+
+function wg_moby_start {
+    _wg_shared_generate_keys
+    _wg_moby_build_module
+    _wg_moby_generate_config
+    _wg_moby_tunnel_open
+}
+
+function wg_moby_stop {
+    _wg_moby_tunnel_close
+}
+
+function _wg_moby_build_module {
+    set -eu
```

Comment:
I really wanted it all in one file so that it is more portable. In its current iteration you need only this one file, and it will bootstrap everything else. From a user perspective, it simplifies deployment to running a single script. I could be swayed on this point, but this is the motivation behind packing it all in a single script. I actually found it surprisingly readable in one file.